viewer/Viewer.cpp

```diff
@@ -784,62 +989,141 @@ namespace Enki
 		updateGL();
 	}
 	
+	void ViewerWidget::keyPressEvent(QKeyEvent* event)
+	{
+		if (event->key() == Qt::Key_F1)
+			showHelp();
+	}
+
 	void ViewerWidget::mousePressEvent(QMouseEvent *event)
 	{
+		// initialization
 		mouseGrabPos = event->pos();
-		/*if (event->button() == Qt::RightButton)
+
+		// change selected object
+		if (event->buttons() & Qt::LeftButton)
 		{
-			mouseGrabbed = true;
-			mouseGrabPos = event->pos();
-		}*/
+			if(selectedObject != pointedObject) trackballView = false;
+			selectedObject = pointedObject;
+		}
+
+		// code button identification for robot
+		int buttonCode = 0;
+		if(event->buttons() & Qt::LeftButton) buttonCode = 1;
+		else if(event->buttons() & Qt::RightButton) buttonCode = 2;
+		else if(event->buttons() & Qt::MiddleButton) buttonCode = 3;
+		else if(event->buttons() & Qt::MidButton) buttonCode = 3;
+
+		// if selected object is a robot call the clicked interaction function
+		Robot* robot = dynamic_cast<Robot*>(pointedObject);
+		if(robot) robot->clickedInteraction(true,buttonCode,pointedPoint.x(),pointedPoint.y(),pointedPoint.z());
 	}
 	
 	void ViewerWidget::mouseReleaseEvent(QMouseEvent * event)
 	{
-		/*if (event->button() == Qt::RightButton)
-			mouseGrabbed = false;*/
+		// enable physics calculation for selected object
+		world->SkipPhysicsObjectsList.erase(selectedObject);
+
+		// code button identification for robot
+		int buttonCode = 0;
+		if(event->buttons() & Qt::LeftButton) buttonCode = 1;
+		else if(event->buttons() & Qt::RightButton) buttonCode = 2;
+		else if(event->buttons() & Qt::MiddleButton) buttonCode = 3;
+		else if(event->buttons() & Qt::MidButton) buttonCode = 3;
+
+		// if selected object is a robot call the clicked interaction function
+		Robot* robot = dynamic_cast<Robot*>(pointedObject);
+		if(robot) robot->clickedInteraction(false,buttonCode,pointedPoint.x(),pointedPoint.y(),pointedPoint.z());
```

Comment:
Enki's coding convention is to put a space between `if` and the opening parenthesis: `if (`.